### PR TITLE
use text-decoration-line instead of text-decoration

### PR DESCRIPTION
**Pull request by o-t-w**
Currently the following code would not lead to the underline being green, which is a bug imo: `class="decoration-green-700 hover:underline hover:underline-offset-4"`

This is because `text-decoration` is a shorthand for `text-decoration-line`, `text-decoration-color`, `text-decoration-style`, and `text-decoration-thickness` so overrides any of those values (if they have been set) to `initial`.

### CODE REVIEW

File: src/corePlugins.js [hunk 1]
```diff
@@ -1491,8 +1491,8 @@ export let corePlugins = {
   padding: createUtilityPlugin('padding', [
     ['p', ['padding']],
     [
-      ['px', ['padding-left', 'padding-right']],
-      ['py', ['padding-top', 'padding-bottom']],
+      ['px', ['padding-inline']],
+      ['py', ['padding-block']],
```
Comment (lukewarlow) [inline src/corePlugins.js:1495]: I would undo this change it's unrelated to the PR

File: src/corePlugins.js [hunk 2]
```diff
@@ -379,8 +379,8 @@ export let corePlugins = {
     [
       ['m', ['margin']],
       [
-        ['mx', ['margin-left', 'margin-right']],
-        ['my', ['margin-top', 'margin-bottom']],
+        ['mx', ['margin-inline']],
+        ['my', ['margin-block']],
```
Comment (lukewarlow) [inline src/corePlugins.js:383]: I would also undo this change it's unrelated to the PR

**Comment by lukewarlow** [comment]
Worth being aware this is technically a breaking change.

**Comment by reinink** [comment]
@o-t-w I like this change.

@lukewarlow can you give an example of how this could break things? Seems pretty unlikely, but maybe I'm missing something?

**Comment by lukewarlow** [comment]
@reinink I agree it's unlikely and someone would have to be relying on what I would agree is buggy behaviour but technically someone could be relying on the fact that text-decoration does override existing values. 

I just thought it was worth pointing out. I'm generally of the opinion these kinds of "breaking changes" are perfectly reasonable though.

**Comment by reinink** [comment]
@o-t-w Hey, I forgot to say thanks when I merged this in earlier. Much appreciated! 🙏
